src/core/Akka/Actor/ActorCell.Children.cs

```diff
@@ -378,15 +377,14 @@ public bool TryGetSingleChild(string name, out IInternalActorRef child)
         /// <returns>TBD</returns>
         protected SuspendReason RemoveChildAndGetStateChange(IActorRef child)
         {
-            var terminating = ChildrenContainer as TerminatingChildrenContainer;
-            if (terminating != null)
+            if (ChildrenContainer is TerminatingChildrenContainer terminating)
             {
                 var newContainer = UpdateChildrenRefs(c => c.Remove(child));
-                if (newContainer is TerminatingChildrenContainer) return null;
+                if (newContainer is TerminatingChildrenContainer) return default(SuspendReason);
                 return terminating.Reason;
             }
             UpdateChildrenRefs(c => c.Remove(child));
-            return null;
+            return default(SuspendReason);
```

Comment:
It’s unclear what this means. We return a struct with all the boolean fields set to false and Status set to zero. Even if it’s what we want, it’s rather implicit and error prone.